rpc/src/main/scala/org/bitcoins/rpc/config/BitcoindInstance.scala

```diff
@@ -26,15 +31,123 @@ object BitcoindInstance {
+                         zmqConfig = zmqConfig)
+  }
+
+  private val DEFAULT_DATADIR =
+    Paths.get(Properties.userHome, ".bitcoin")
+
+  private val DEFAULT_CONF_FILE = DEFAULT_DATADIR.resolve("bitcoin.conf")
+
+  def fromDatadir(datadir: File = DEFAULT_DATADIR.toFile): BitcoindInstance = {
+    require(datadir.exists, s"${datadir.getPath} does not exist!")
+    require(datadir.isDirectory, s"${datadir.getPath} is not a directory!")
+
+    val file = Paths.get(datadir.getAbsolutePath, "bitcoin.conf").toFile
+    fromConfigFile(file)
+  }
+
+  def fromConfigFile(
+      file: File = DEFAULT_CONF_FILE.toFile): BitcoindInstance = {
+    require(file.exists, s"${file.getPath} does not exist!")
+    require(file.isFile, s"${file.getPath} is not a file!")
+
+    val config = ConfigFactory.parseFile(file)
+
+    // skip .resolve because Bitcoin Core only uses simple INI-like notation,
+    // not environment variable substitution
+    fromConfig(config)
+  }
+
+  def fromConfig(
+      config: Config,
+      datadir: File = DEFAULT_DATADIR.toFile
+  ): BitcoindInstance = {
+    val network = getNetwork(config)
+
+    val uri = getUri(config, network)
+    val rpcUri = getRpcUri(config, network)
+
+    val username = config.getString("rpcuser")
+    val password = config.getString("rpcpassword")
+    val authCredentials =
+      BitcoindAuthCredentials(username = username,
+                              password = password,
+                              rpcPort = rpcUri.getPort,
+                              datadir = datadir)
+
+    BitcoindInstance(network,
+                     uri,
+                     rpcUri,
+                     authCredentials,
+                     zmqConfig = ZmqConfig.fromConfig(config))
+  }
+
+  private def isSet(config: Config, path: String): Boolean = {
+    Try(config.getInt(path))
+      .map(_ == 1)
+      .getOrElse(false)
+  }
+
+  private def getNetwork(config: Config): BitcoinNetwork = {
+    val isTestnet = isSet(config, path = "testnet")
+    val isRegTest = isSet(config, path = "regtest")
+
+    (isRegTest, isTestnet) match {
+      case (true, true) =>
+        throw new IllegalArgumentException(
+          """"Cannot set both "regtest" and "testnet" options""")
+      case (true, false)  => RegTest
+      case (false, true)  => TestNet3
+      case (false, false) => MainNet
+    }
+  }
+
+  private def getUri(config: Config, network: NetworkParameters): URI = {
+    val port = Try(config.getInt("port")).getOrElse(network.port)
+    val host = Try(config.getString("bind")).getOrElse("localhost")
+    val uriT = Try {
+      new URI(s"http://$host:$port")
+    }
+
+    uriT match {
+      case Success(uriSuccess) => uriSuccess
+      case Failure(exception) =>
+        throw new IllegalArgumentException(
+          s"Could not construct URI from host $host and port $port",
+          exception)
+    }
+
   }
+
+  private def getRpcUri(config: Config, network: NetworkParameters) = {
```

Comment:
type annotation on method definitition

Reply:
Maybe just return `Try[URI]` rather than throwing?

Reply:
We would have to throw somewhere though, if invalid config opts are specified

Reply:
Also I don't mind adding the type annotation, but I generally think that's overkill on private methods. What's your opinion on that?